os/net/netdev/netdev_dhcpc.c

```diff
@@ -0,0 +1,285 @@
+	inaddr.sin_port = 0;
+	memcpy(&inaddr.sin_addr, addr, sizeof(struct in_addr));
+
+	_ioctl_setipv4addr(&ip4_addr_get_u32(ip_2_ip4(&gw)), (struct sockaddr *)&inaddr);
+	ipaddr = dev->ip_addr;
+	netmask = dev->netmask;
+	netifapi_netif_set_addr(dev, ip_2_ip4(&ipaddr), ip_2_ip4(&netmask), ip_2_ip4(&gw));
+
+	netifapi_netif_set_up(dev);
+	return 0;
+}
+
+int _netlib_get_ipv4addr(FAR const char *ifname, FAR struct in_addr *addr)
+{
+	struct netif *dev = netdev_findbyname(ifname);
+	if (!dev) {
+		return -1;
+	}
+
+	struct sockaddr_in inaddr;
+
+	ioctl_getipv4addr((struct sockaddr *)&inaddr, ip4_addr_get_u32(ip_2_ip4(&dev->ip_addr)));
+
+	memcpy(addr, &inaddr.sin_addr.s_addr, sizeof(struct in_addr));
+
+	return 0;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: dhcp_client_start
+ ****************************************************************************/
+int netdev_dhcp_client_start(const char *intf)
+{
+	ndbg("[DHCPC] LWIP DHCPC started\n");
+	struct netif *cur_netif;
+	cur_netif = netif_find(intf);
+	if (cur_netif == NULL) {
+		ndbg("[DHCPC] No network interface for dhcpc\n");
+		return -1;
+	}
+	int32_t timeleft = CONFIG_LWIP_DHCPC_TIMEOUT;
+	struct in_addr local_ipaddr;
+	struct in_addr local_netmask;
+	struct in_addr local_gateway;
+
+	struct in_addr ip_check;
+
+	/* Initialize dhcp structure if exists */
+	if (netif_dhcp_data(cur_netif)) {
+		netifapi_dhcp_stop(cur_netif);
+	}
+
+	local_ipaddr.s_addr = IPADDR_ANY;
+	local_netmask.s_addr = IPADDR_BROADCAST;
+	local_gateway.s_addr = IPADDR_ANY;
+	DHCPC_SET_IP4ADDR(intf, local_ipaddr, local_netmask, local_gateway);
+	err_t res = netifapi_dhcp_start(cur_netif);
+	if (res) {
+		ndbg("[DHCPC] dhcpc_start failure %d\n", res);
+		return -1;
+	}
+	ndbg("[DHCPC] dhcpc_start success, waiting IP address (timeout %d secs)\n", timeleft);
+	while (netifapi_dhcp_address_valid(cur_netif) != 0) {
+		usleep(100000);
+		timeleft -= 100;
+		if (timeleft <= 0) {
+			ndbg("[DHCPC] dhcpc_start timeout\n");
+			netifapi_dhcp_stop(cur_netif);
+			return -1;
+		}
+	}
+	int ret = _netlib_get_ipv4addr(intf, &ip_check);
+	if (ret == -1) {
+		ndbg("[DHCPC] fail to get IP address\n");
+		netifapi_dhcp_stop(cur_netif);
+		return -1;
+	}
+	ndbg("[DHCPC] dhcpc_start - got IP address %s\n", inet_ntoa(ip_check));
+	return OK;
+}
+
+/****************************************************************************
+ * Name: dhcp_client_stop
+ ****************************************************************************/
+void netdev_dhcp_client_stop(const char *intf)
+{
+	struct in_addr in = { .s_addr = INADDR_NONE };
+	DHCPC_SET_IP4ADDR(intf, in, in, in);
+	struct netif *cur_netif;
+	cur_netif = netif_find(intf);
+	if (cur_netif == NULL) {
+		ndbg("[DHCPC] No network interface for dhcpc\n");
```

Comment:
Why do you need prefix `[DHCPC]` with dbg?

Reply:
It is easy to recognize DHCPC log among many log msgs. Thanks for the comment.